lib/money_column/active_record_hooks.rb

```diff
@@ -48,7 +48,12 @@ def write_money_attribute(column, money)
         return self[column] = nil
       end
 
-      currency_raw_source = options[:currency] || (send(options[:currency_column]) rescue nil)
+      currency_raw_source = begin
+        options[:currency] || send(options[:currency_column]) || money.currency
```

Comment:
I think it should - because a couple of lines down we check if the currency passed in is allowed, aka does it match a specified currency already on that column. I.e. default is 'CAD` but you pass in 'GBP', it should fail.